sycl/source/kernel_bundle.cpp

```diff
@@ -296,20 +296,17 @@ std::vector<kernel_id> get_kernel_ids() {
 }
 
 bool is_compatible(const std::vector<kernel_id> &KernelIDs, const device &Dev) {
-  for (const auto &KernelId : KernelIDs) {
-    const detail::RTDeviceBinaryImage &Img =
-        detail::ProgramManager::getInstance().getDeviceImage(
-            detail::OSUtil::ExeModuleHandle, KernelId.get_name(), context(Dev),
-            Dev);
-    const detail::RTDeviceBinaryImage::PropertyRange &ARange =
-        Img.getDeviceRequirements();
-    for (detail::RTDeviceBinaryImage::PropertyRange::ConstIterator It :
-         ARange) {
+  using namespace detail;
+  std::set<RTDeviceBinaryImage *> BinImages;
+  ProgramManager::getInstance().getRawDeviceImages(KernelIDs, BinImages);
+  for (RTDeviceBinaryImage *Img : BinImages) {
+    const RTDeviceBinaryImage::PropertyRange &PropRange =
+        Img->getDeviceRequirements();
+    for (RTDeviceBinaryImage::PropertyRange::ConstIterator It : PropRange) {
       using namespace std::literals;
       if ((*It)->Name != "aspects"sv)
         continue;
-      detail::ByteArray Aspects =
-          detail::DeviceBinaryProperty(*It).asByteArray();
+      ByteArray Aspects = DeviceBinaryProperty(*It).asByteArray();
```

Comment:
Are the changes in this file just cosmetic?

Reply:
No, it's a functional change. I don't know why, the previous version was working fine with the E2E test, but not with this unit. It seems like another map (m_KernelSets) that was used in `getDeviceImage() -> getKernelSetId()` is not initialized there. But this works good for both cases.

Reply:
I see, thanks!